backend/dataall/core/permissions/db/resource_policy/resource_policy_repositories.py

```diff
@@ -71,16 +70,35 @@ def has_group_resource_permission(
         else:
             return policy
 
-    @staticmethod
-    def find_resource_policy(session, group_uri: str, resource_uri: str) -> ResourcePolicy:
-        resource_policy = (
-            session.query(ResourcePolicy)
-            .filter(
-                and_(
-                    ResourcePolicy.principalId == group_uri,
-                    ResourcePolicy.resourceUri == resource_uri,
-                )
-            )
-            .first()
+    def query_all_resource_policies(session, group_uri: str, resource_uri: str, resource_type: str = None):
+        resource_policy = session.query(ResourcePolicy).filter(
+            ResourcePolicy.principalId == group_uri,
+            ResourcePolicy.resourceUri == resource_uri,
         )
+        if group_uri is not None:
+            resource_policy = resource_policy.filter(
+                ResourcePolicy.principalId == group_uri,
+            )
```

Comment:
We are filtering on `ResourcePolicy.principalId == group_uri` on L75 and L79-81

Can we remove L75 and keep logic in L78-81?

Reply:
It's typo. Surely it's not necessary in line 75